### ISSUE TITLE
Squiz.WhiteSpace.ScopeKeywordSpacing false positive for static return type

### ISSUE THREAD

**Issue by mathieutu** [squizlabs/PHP_CodeSniffer#3188]
**Describe the bug**
Php8 allows static as a valid return type. I've seen the PHPCS PR which allows that.
However I still have an issue with the `Squiz.WhiteSpace.ScopeKeywordSpacing.Incorrect` rule.

**Code sample**
```php
public static function fCreate($attributes = []): static
{
    return static::factory()->create($attributes);
}
```

**Error**
```
 35 | ERROR | [x] Scope keyword "static" must be followed by a single space; found 0 (Squiz.WhiteSpace.ScopeKeywordSpacing.Incorrect)
 43 | ERROR | [x] Scope keyword "static" must be followed by a single space; found newline
    |       |     (Squiz.WhiteSpace.ScopeKeywordSpacing.Incorrect)
 ```

**Expected behavior**
Phpcs should not emit an error for a space after static.
And actually `static` as a return type should be considerated by phpcs as a Scope Keyword.

**Versions:**
 - OS: MacOS 10.15.7
 - PHP: 8
 - PHPCS: 3.5.8
 - Standard: [PSR2, PSR12]

**Comment by jrfnl** [comment]
Thanks for reporting this. I can not reproduce the first error. The second one (line 43) I _can_ reproduce with your code sample.

PR #3189 should fix this. Testing appreciated.

**Comment by mathieutu** [comment]
Thanks for this so fast response!

Indeed, the error line 35 was actually caused by 
```php
    public static function myMethod(): static|\Closure
```

Do you think youre fix will cover it too?

**Comment by mathieutu** [comment]
@jrfnl I can confirm it's ok, both problems solved!

**Comment by jrfnl** [comment]
Thanks for testing @mathieutu! Glad to see the confirmation that this will now be fixed once the PR is merged.
